views/default/create_packages.pdt

```diff
@@ -0,0 +1,446 @@
+                                        <?php $this->Form->fieldText('renew_price_markup', $this->Html->ifSet($vars->renew_price_markup), ['class' => 'stretch']);?>
+                                    </td>
+                                    <td class="large" colspan="2">
+                                        <?php
+                                        $this->Form->fieldCheckbox('price_enable_renews_all', '1', ($this->Html->ifSet($vars->price_enable_renews_all, '1') == '1'), ['id' => 'price_enable_renews_all']);
+                                        $this->Form->label($this->_('Namesilo.manage.create_packages.field_price_enable_renews_all', true), 'price_enable_renews_all', ['class' => 'inline']);
+                                        ?>
+                                    </td>
+                                </tr>
+                            </tbody>
+                        </table>
+                        <ul>
+                            <li class="pad_top">
+                                <?php
+                                $this->Form->fieldCheckbox('taxable', '1', ($this->Html->ifSet($vars->taxable, '0') == '1'), ['id' => 'taxable']);
+                                $this->Form->label($this->_('Namesilo.manage.create_packages.field_taxable', true), 'taxable', ['class' => 'inline']);
+                                ?>
+                            </li>
+                        </ul>
+                    </li>
+                    <li>
+                        <table class="table">
+                            <thead>
+                                <tr class="heading_row">
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.tld');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.currency');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.previous_registration_price');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.previous_renewal_price');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.current_registration_price');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.current_renewal_price');?></td>
+                                    <td><?php $this->_('Namesilo.manage.create_packages_heading.price');?></td>
+                                    <td class="last"><?php $this->_('Namesilo.manage.create_packages_heading.renew_price');?></td>
+                                </tr>
+                            </thead>
+                            <tbody>
+                                <?php
+                                $i = 0;
+                                foreach ($tlds as $tld => $pricing) {
+                                    foreach ($currencies as $currency) {
```

Comment:
These loops create hundreds or thousands of rows. That is really slow (for me it took 3 full minutes to load).  We need a better way of doing this.  Perhaps you could load the pricing rows for a tlds via ajax when you check it.  To prevent refetching prices from the api you could probably cache the results either in a file or the database.